test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -263,25 +264,64 @@ private static void getFailureReason() {
      * testWindow.
      *
      * @param testWindow test window that the test is created
-     * @param position  position can be either HORIZONTAL (both test
-     *                  instruction frame and test window as arranged
-     *                  side by side) or VERTICAL (both test instruction
-     *                  frame and test window as arranged up and down)
+     * @param position  position can either be:
+     *                  HORIZONTAL - both test instruction frame and test window
+     *                  as arranged side by side.
+     *                  VERTICAL - both test instruction frame and test window
+     *                  as arranged up and down.
+     *                  TOP_LEFT_CORNER - test instruction frame positioned at
+     *                  top left corner with main test window beside it.
      */
     public static void positionTestWindow(Window testWindow, Position position) {
         Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+
         if (position.equals(Position.HORIZONTAL)) {
             int newX = ((screenSize.width / 2) - frame.getWidth());
             frame.setLocation(newX, frame.getY());
 
-            testWindow.setLocation((frame.getLocation().x + frame.getWidth() + 5), frame.getY());
+            testWindow.setLocation((frame.getLocation().x + frame.getWidth() + 5),
+                    frame.getY());
         } else if (position.equals(Position.VERTICAL)) {
             int newY = ((screenSize.height / 2) - frame.getHeight());
             frame.setLocation(frame.getX(), newY);
 
             testWindow.setLocation(frame.getX(),
                     (frame.getLocation().y + frame.getHeight() + 5));
         }
+        else if (position.equals(Position.TOP_LEFT_CORNER)) {
+            frame.setLocation(0,0);
```

Comment:
I am very sceptical of this. With a toolbar on the left and a menu bar on the top of the screen I bet you
are neither at x=0 or y=0.
And this code is not allowing the desktop to reposition it and report the updated REAL position back to you
The result will be overlapping windows.
I'm actually dubious of the existing code for much the same reason but it just likely hasn't been seen as a problem there because positioning in the middle of the screen usually gets you where you asked for.

Reply:
@prrace Thank you for reviewing. I do see your point - when taskbars and menu bars are located on top or left side of the screen, they would overlap the test instruction frame.  

Two possible solutions:

1. Going with a fixed amount of offset on where to position the top-left corner of the instruction frame window. Probably this would not be ideal considering different screen settings and platforms.

2. The other approach would be to use [GraphicsEnvironment.getMaximumWindowBounds()](https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/java/awt/GraphicsEnvironment.html#getMaximumWindowBounds()) to get the displayable window size and position the instruction frame accordingly ?

Reply:
More than that - this code needs to call setlocation, and do "whatever" to make sure that is actually pushed to the "window manager" and the REAL location of the window come back. The way the code is written it sets a couple of vars and then reads back its own values .. no chance of finding the REAL location

Reply:
In the recent update the following points are addressed:

- Screen insets are used to account of taskbar position and the placement of the instruction frame at top-left corner
- Position option docs updated

Changes regarding pushing the latest position to window manager are being evaluated.

Reply:
Following is a proposed solution to push the latest position changes to the window manager.

Robot.waitForIdle() are added between setLocation() and the subsequent getLocation() calls. This ensures any pending location updates are pushed to the window manager before the new location of the frame is in turn used to position the testWindow.

In multiple test window cases ONLY the test instruction frame and the main/root test window are positioned using `positionTestWindow()`, the rest of the windows are positioned within EDT at test-level.

**Impact of the proposed fix:** 
With this change `positionTestWindow()` can no longer be called on EDT at test-level, hence any existing manual tests calling this method on EDT are required to be changed. 

For example: In this particular test, the following line `PassFailJFrame.positionTestWindow(frame,PassFailJFrame.Position.HORIZONTAL)` should be called after `createAndShowGUI()`

https://github.com/openjdk/jdk/blob/d333df8ff59c6242171da1a6d02d05e6a8e65c9d/test/jdk/javax/swing/JComboBox/JComboBoxActionEvent.java#L75
 
```
public static void positionTestWindow(Window testWindow, Position position) throws AWTException {

        if (isEventDispatchThread()) {
            throw new IllegalStateException("positionTestWindow() should not be called on EDT");
        }

        robot = new Robot();
        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();

        if (position.equals(Position.HORIZONTAL)) {
            SwingUtilities.invokeLater(() -> {
                int newX = ((screenSize.width / 2) - frame.getWidth());
                frame.setLocation(newX, frame.getY());
           });

            robot.waitForIdle();
            robot.delay(300);

            SwingUtilities.invokeLater(() -> testWindow.setLocation((frame.getX() +
                                             frame.getWidth() + 5), frame.getY()));

        } else if (position.equals(Position.VERTICAL)) {
            SwingUtilities.invokeLater(() -> {
                int newY = ((screenSize.height / 2) - frame.getHeight());
                frame.setLocation(frame.getX(), newY);
            });

            robot.waitForIdle();
            robot.delay(300);

            SwingUtilities.invokeLater(() -> testWindow.setLocation(frame.getX(),
                                            (frame.getY() + frame.getHeight() + 5)));

        } else if (position.equals(Position.TOP_LEFT_CORNER)) {
            SwingUtilities.invokeLater(() -> {
                GraphicsConfiguration gc = GraphicsEnvironment.getLocalGraphicsEnvironment()
                        .getDefaultScreenDevice().getDefaultConfiguration();
                Insets screenInsets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
                frame.setLocation(screenInsets.left, screenInsets.top);
            });

            robot.waitForIdle();
            robot.delay(300);

            SwingUtilities.invokeLater(() -> testWindow.setLocation((frame.getX() +
                   frame.getWidth() + 5), frame.getY()));
        }
    }
```

Reply:
We should discuss that impact. 
This method is useful to simplify the logic of individual tests but is it now becoming awkward to use ?
Or is it something that we should have required from the beginning  ?
And why invokeLater and not invokeAndWait ? I'd have expected Later to not work for this case ..

Reply:
The interleaving does seem to make the code more complex. And at test-level, it causes some confusion as few method need to called within EDT and few out of EDT. 

I wanted to check if there was a simpler, much clearer approach and evaluate possible options before proceeding.

I might have gone with invokeLater, since we are calling waitForIdle() later  and it waits until the queue is empty. But I think invokeAndWait would be a better option here.

Reply:
@prrace Updated the PR  - added Toolkit.sync and Thread.sleep. Additionally, added few more changes as mentioned here - https://github.com/openjdk/jdk/pull/9525#issuecomment-1205915695 & https://github.com/openjdk/jdk/pull/9525#issuecomment-1206690566